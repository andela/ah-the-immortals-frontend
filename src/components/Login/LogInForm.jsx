import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const LogInForm = ({ handleSignInSubmit, handleChange }) => (
  <form onSubmit={handleSignInSubmit} className="modal-form">
    <label htmlFor="email">
      <input type="email" required className="form-control" name="email" onChange={handleChange} placeholder="Email" />
    </label>
    <label htmlFor="password">
      <input type="password" required className="form-control" name="password" onChange={handleChange} placeholder="Password" />
    </label>
    <Button type="submit" variant="info" className="form-control btn" value="SIGN IN">
        SIGN IN
    </Button>
    <div className="reset-div">
      <Link to="/reset" className="passreset-link">Forgot Password?</Link>
    </div>
  </form>
);

LogInForm.propTypes = {
  handleSignInSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default LogInForm;
