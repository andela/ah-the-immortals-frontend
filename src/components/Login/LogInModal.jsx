import React from 'react';
import { Container , Modal } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFacebook, faTwitter, faGooglePlus } from '@fortawesome/free-brands-svg-icons';
import { library } from '@fortawesome/fontawesome-svg-core';
import PropTypes from 'prop-types';
import '../../styles/style.css';
import '../../styles/App.css'; 
import LogInForm from './LogInForm';

library.add(faFacebook, faTwitter, faGooglePlus);

const LogInModal = ({
  closeModal,
  signInShow,
  handleChange,
  handleSignInSubmit
}) => {
  return (
    <Modal show={signInShow} onHide={closeModal}>
      <Modal.Header closeButton>
      </Modal.Header>
      <Modal.Body className="modal-body">
        <div className="shadow-lg p-3 mb-5 bg-white rounded form-div">
          <p>Sign In</p>
          <LogInForm
            handleChange={handleChange}
            handleSignInSubmit={handleSignInSubmit}
          />
          <h2><span className="bg-light">OR</span></h2>
          <div className="round-icon">
            <FontAwesomeIcon icon={['fab', 'twitter']} className="twitter" />
            <FontAwesomeIcon icon={['fab', 'facebook']} className="facebook" />
            <FontAwesomeIcon icon={['fab', 'google-plus']} className="google" />
          </div>
        </div>
      </Modal.Body>
    </Modal>
  );
};

LogInModal.propTypes = {
  signInShow: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSignInSubmit: PropTypes.func.isRequired
};

export default LogInModal;
