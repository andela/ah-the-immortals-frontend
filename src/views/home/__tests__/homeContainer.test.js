import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import { CloseButton } from 'react-bootstrap';
import Form from '../../../components/signup/FormComponent';
import SignupModal from '../../../components/signup/SignupComponent';
import Home from '../HomeContainer';
import LogInModal from '../../../components/Login/LogInModal';
import LogInForm from '../../../components/Login/LogInForm';

describe('Tests Home container', () => {
  const store = configureStore([thunk])({
    signup: { user: {}, errors: {} }
  });
  const history = createBrowserHistory();
  const wrapper = mount(
    <Router history={history}>
      <Provider store={store}>
        <Home />
      </Provider>
    </Router>
  );

  it('Tests mounting of component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('opens modal when button is clicked', () => {
    const signupBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Join now'));

    signupBtn.simulate('click', {
      preventDefault: jest.fn()
    });
    expect(wrapper.find(SignupModal).prop('show')).toEqual(true);
  });

  it('handles input change in form', () => {
    const signupBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Join now'));

    signupBtn.simulate('click', {
      preventDefault: jest.fn()
    });
    const form = wrapper.find(Form);
    const input = form.find('input[name="username"]');
    input.simulate('change', {
      preventDefault: jest.fn(),
      target: {
        value: 'brian',
        name: 'username'
      }
    });
    expect(input.length).toEqual(1);
  });

  it('handles submit event', () => {
    const signupBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Join now'));

    signupBtn.simulate('click', {
      preventDefault: jest.fn()
    });

    
    const form = wrapper.find(Form);
    const submitBtn = form.find('input[type="submit"]');

    submitBtn.simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(submitBtn.length).toEqual(1);
  });

  it('handles close modal', () => {
    const signupBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Join now'));

    signupBtn.simulate('click', {
      preventDefault: jest.fn()
    });

    const modal = wrapper.find(SignupModal);
    const closeBtn = modal.find(CloseButton);

    closeBtn.simulate('click');
    expect(closeBtn.length).toEqual(1);
  });

  it('opens sign in modal when button is clicked', () => {
    const signInBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Sign In'));

    signInBtn.simulate('click', {
      preventDefault: jest.fn()
    });
    expect(wrapper.find(LogInModal).prop('signInShow')).toEqual(true);
  });

  it('handles sign in submit', () => {
    const signInBtn = wrapper.findWhere(n => n.type() === 'button' && n.contains('Sign In'));

    signInBtn.simulate('click', {
      preventDefault: jest.fn()
    });

    const loginForm = wrapper.find(LogInForm);
    const submitBtn = loginForm.find('button[type="submit"]');

    submitBtn.simulate('submit', {
      preventDefault: jest.fn()
    });
    expect(submitBtn.length).toEqual(1);
  });
});
