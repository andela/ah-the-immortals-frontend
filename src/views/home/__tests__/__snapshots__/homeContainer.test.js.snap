// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Home container Tests mounting of component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(Home)>
      <Home
        signInAction={[Function]}
        signup={[Function]}
        signupdata={
          Object {
            "errors": Object {},
            "user": Object {},
          }
        }
      >
        <div>
          <ToastContainer
            autoClose={5000}
            bodyClassName={null}
            className={null}
            closeButton={
              <CloseButton
                ariaLabel="close"
              />
            }
            closeOnClick={true}
            draggable={true}
            draggablePercent={80}
            hideProgressBar={false}
            newestOnTop={false}
            pauseOnFocusLoss={true}
            pauseOnHover={true}
            position="top-right"
            progressClassName={null}
            progressStyle={null}
            role="alert"
            rtl={false}
            style={null}
            toastClassName={null}
            transition={[Function]}
          >
            <div
              className="Toastify"
            />
          </ToastContainer>
          <Navigationbar
            showModal={[Function]}
            signInShow={[Function]}
          >
            <div>
              <Bootstrap(undefined)
                className="homenav fixed-top"
              >
                <ForwardRef
                  bsPrefix="navbar"
                  className="homenav fixed-top"
                  innerRef={null}
                >
                  <Uncontrolled(Navbar)
                    bsPrefix="navbar"
                    className="homenav fixed-top"
                    innerRef={null}
                  >
                    <Navbar
                      bsPrefix="navbar"
                      className="homenav fixed-top"
                      collapseOnSelect={false}
                      expand={true}
                      onToggle={[Function]}
                      variant="light"
                    >
                      <nav
                        className="homenav fixed-top navbar navbar-expand navbar-light"
                      >
                        <NavbarBrand>
                          <span
                            className="navbar-brand"
                          >
                            <b
                              className="brand"
                            >
                              Authors Haven
                            </b>
                          </span>
                        </NavbarBrand>
                        <NavbarCollapse>
                          <Collapse
                            appear={false}
                            dimension="height"
                            getDimensionValue={[Function]}
                            in={false}
                            mountOnEnter={false}
                            timeout={300}
                            unmountOnExit={false}
                          >
                            <Transition
                              addEndListener={[Function]}
                              appear={false}
                              aria-expanded={null}
                              enter={true}
                              exit={true}
                              in={false}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={300}
                              unmountOnExit={false}
                            >
                              <div
                                aria-expanded={null}
                                className="navbar-collapse collapse"
                              >
                                <Nav
                                  className="justify-content-end"
                                  fill={false}
                                  justify={false}
                                  style={
                                    Object {
                                      "width": "100%",
                                    }
                                  }
                                >
                                  <ContextTransform(AbstractNav)
                                    as="div"
                                    className="justify-content-end navbar-nav"
                                    onSelect={[Function]}
                                    style={
                                      Object {
                                        "width": "100%",
                                      }
                                    }
                                  >
                                    <AbstractNav
                                      as="div"
                                      className="justify-content-end navbar-nav"
                                      onSelect={[Function]}
                                      parentOnSelect={[Function]}
                                      style={
                                        Object {
                                          "width": "100%",
                                        }
                                      }
                                    >
                                      <div
                                        className="justify-content-end navbar-nav"
                                        onKeyDown={[Function]}
                                        style={
                                          Object {
                                            "width": "100%",
                                          }
                                        }
                                      >
                                        <NavItem>
                                          <div
                                            className="nav-item"
                                          >
                                            <Button
                                              active={false}
                                              className="buttons"
                                              disabled={false}
                                              onClick={[Function]}
                                              type="button"
                                              variant="danger"
                                            >
                                              <button
                                                className="buttons btn btn-danger"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                Join now
                                              </button>
                                            </Button>
                                          </div>
                                        </NavItem>
                                        <NavItem>
                                          <div
                                            className="nav-item"
                                          >
                                            <Button
                                              active={false}
                                              className="buttons"
                                              disabled={false}
                                              onClick={[Function]}
                                              type="button"
                                              variant="outline-light"
                                            >
                                              <button
                                                className="buttons btn btn-outline-light"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                Sign In
                                              </button>
                                            </Button>
                                          </div>
                                        </NavItem>
                                      </div>
                                    </AbstractNav>
                                  </ContextTransform(AbstractNav)>
                                </Nav>
                              </div>
                            </Transition>
                          </Collapse>
                        </NavbarCollapse>
                      </nav>
                    </Navbar>
                  </Uncontrolled(Navbar)>
                </ForwardRef>
              </Bootstrap(undefined)>
            </div>
          </Navigationbar>
          <Container
            className="homepage"
            fluid={false}
          >
            <div
              className="homepage container"
            >
              <HomePage
                closeModal={[Function]}
                handleChange={[Function]}
                handleSignInSubmit={[Function]}
                handleSubmit={[Function]}
                show={false}
                showModal={[Function]}
                signInShow={false}
              >
                <div
                  className="columns"
                >
                  <Bootstrap(Row)>
                    <Row
                      bsPrefix="row"
                      noGutters={false}
                    >
                      <div
                        className="row"
                      >
                        <Col
                          className="col"
                        >
                          <div
                            className="col col"
                          >
                            <Container
                              fluid={false}
                            >
                              <div
                                className="container"
                              >
                                <h1
                                  className="homepage"
                                >
                                  Welcome to your articles Community
                                </h1>
                                <br />
                                <Form
                                  inline={false}
                                >
                                  <form
                                    className=""
                                  >
                                    <FormControl
                                      className="font-awesome"
                                      placeholder="  Search for articles"
                                      type="text"
                                    >
                                      <input
                                        className="font-awesome form-control"
                                        placeholder="  Search for articles"
                                        type="text"
                                      />
                                    </FormControl>
                                  </form>
                                </Form>
                              </div>
                            </Container>
                            <Container
                              className="home"
                              fluid={false}
                            >
                              <div
                                className="home container"
                              >
                                <p>
                                  <b>
                                    Who is Authors Haven for?
                                  </b>
                                </p>
                                <p>
                                  Anyone looking to inspire and be inspired by ideas
                                </p>
                                <p>
                                  Join fellow authors and enthusiasts in Authors Haven
                                </p>
                                <Button
                                  active={false}
                                  disabled={false}
                                  onClick={[Function]}
                                  type="button"
                                  variant="primary"
                                >
                                  <button
                                    className="btn btn-primary"
                                    disabled={false}
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    Get Started
                                  </button>
                                </Button>
                                <SignupModal
                                  closeModal={[Function]}
                                  handleChange={[Function]}
                                  handleSubmit={[Function]}
                                  show={false}
                                >
                                  <Bootstrap(Modal)
                                    onHide={[Function]}
                                    show={false}
                                  >
                                    <Modal
                                      animation={true}
                                      autoFocus={true}
                                      backdrop={true}
                                      bsPrefix="modal"
                                      dialogAs={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "displayName": "ModalDialog",
                                          "render": [Function],
                                        }
                                      }
                                      enforceFocus={true}
                                      keyboard={true}
                                      manager={
                                        BootstrapModalManager {
                                          "adjustAndStore": [Function],
                                          "containers": Array [],
                                          "data": Array [],
                                          "handleContainerOverflow": true,
                                          "hideSiblingNodes": true,
                                          "modals": Array [],
                                          "restore": [Function],
                                          "scrollbarSize": 0,
                                        }
                                      }
                                      onHide={[Function]}
                                      restoreFocus={true}
                                      show={false}
                                    >
                                      <Modal
                                        autoFocus={true}
                                        backdrop={true}
                                        backdropTransition={[Function]}
                                        className="modal"
                                        containerClassName="modal-open"
                                        enforceFocus={true}
                                        keyboard={true}
                                        manager={
                                          BootstrapModalManager {
                                            "adjustAndStore": [Function],
                                            "containers": Array [],
                                            "data": Array [],
                                            "handleContainerOverflow": true,
                                            "hideSiblingNodes": true,
                                            "modals": Array [],
                                            "restore": [Function],
                                            "scrollbarSize": 0,
                                          }
                                        }
                                        onClick={[Function]}
                                        onEnter={[Function]}
                                        onEntering={[Function]}
                                        onExited={[Function]}
                                        onHide={[Function]}
                                        onMouseUp={[Function]}
                                        renderBackdrop={[Function]}
                                        restoreFocus={true}
                                        role="dialog"
                                        show={false}
                                        style={Object {}}
                                        transition={[Function]}
                                      />
                                    </Modal>
                                  </Bootstrap(Modal)>
                                </SignupModal>
                                <LogInModal
                                  closeModal={[Function]}
                                  handleChange={[Function]}
                                  handleSignInSubmit={[Function]}
                                  signInShow={false}
                                >
                                  <Bootstrap(Modal)
                                    onHide={[Function]}
                                    show={false}
                                  >
                                    <Modal
                                      animation={true}
                                      autoFocus={true}
                                      backdrop={true}
                                      bsPrefix="modal"
                                      dialogAs={
                                        Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "displayName": "ModalDialog",
                                          "render": [Function],
                                        }
                                      }
                                      enforceFocus={true}
                                      keyboard={true}
                                      manager={
                                        BootstrapModalManager {
                                          "adjustAndStore": [Function],
                                          "containers": Array [],
                                          "data": Array [],
                                          "handleContainerOverflow": true,
                                          "hideSiblingNodes": true,
                                          "modals": Array [],
                                          "restore": [Function],
                                          "scrollbarSize": 0,
                                        }
                                      }
                                      onHide={[Function]}
                                      restoreFocus={true}
                                      show={false}
                                    >
                                      <Modal
                                        autoFocus={true}
                                        backdrop={true}
                                        backdropTransition={[Function]}
                                        className="modal"
                                        containerClassName="modal-open"
                                        enforceFocus={true}
                                        keyboard={true}
                                        manager={
                                          BootstrapModalManager {
                                            "adjustAndStore": [Function],
                                            "containers": Array [],
                                            "data": Array [],
                                            "handleContainerOverflow": true,
                                            "hideSiblingNodes": true,
                                            "modals": Array [],
                                            "restore": [Function],
                                            "scrollbarSize": 0,
                                          }
                                        }
                                        onClick={[Function]}
                                        onEnter={[Function]}
                                        onEntering={[Function]}
                                        onExited={[Function]}
                                        onHide={[Function]}
                                        onMouseUp={[Function]}
                                        renderBackdrop={[Function]}
                                        restoreFocus={true}
                                        role="dialog"
                                        show={false}
                                        style={Object {}}
                                        transition={[Function]}
                                      />
                                    </Modal>
                                  </Bootstrap(Modal)>
                                </LogInModal>
                              </div>
                            </Container>
                          </div>
                        </Col>
                        <Col
                          className="homepageimg"
                        >
                          <div
                            className="homepageimg col"
                          >
                            <img
                              alt=""
                              height="400"
                              src="test-file-stub"
                              width="450"
                            />
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </Bootstrap(Row)>
                </div>
              </HomePage>
            </div>
          </Container>
          <Footer>
            <footer
              className="bg-light text-dark mt-4 fixed-bottom"
            >
              <div
                className="container-fluid py-2"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-md-3 footmargin"
                  >
                    Authors Haven
                     
                    <span
                      className="small"
                    >
                      <br />
                      <a
                        href="#null"
                      >
                        Terms and Conditions
                      </a>
                    </span>
                  </div>
                  <div
                    className="col-md-8 text-right small align-self-end"
                  >
                    Copyright ©
                     
                    2019
                     
                    Immortals
                  </div>
                </div>
              </div>
            </footer>
          </Footer>
        </div>
      </Home>
    </Connect(Home)>
  </Provider>
</Router>
`;
